src/components/pokedex.jsx

```diff
@@ -1,17 +1,64 @@
+import { usePokemon } from '../context/pokemonContext';
+import { useState } from 'react';
+
 export default function Pokedex() {
+    const pokemons = usePokemon();
+    const [typedName, setTypedName] = useState('');
+    const [imageUrl, setImageUrl] = useState(
+        'https://raw.githubusercontent.com/fanzeyi/pokemon.json/master/images/150.png',
+    );
+    let [isError, setIsError] = useState(false);
+
+    let handleChange = ({ target }) => {
+        setTypedName(target.value);
+        setIsError(false);
+    };
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+
+        pokemons.map((pokemon) => {
+            if (pokemon.name.toLowerCase() === typedName.toLowerCase()) {
+                setImageUrl(pokemon.imageUrl);
+            }
+        });
+
+        //Search for pokemon data and return an error if "typedName" is not matching any of the pokemons.
+        setIsError(
+            !pokemons.some(
+                (pokemon) =>
+                    pokemon.name.toLowerCase() === typedName.toLowerCase(),
+            ),
+        );
+    };
```

Comment:
I think we could make this abit easier to reason with if we used Array.find, to see if there is a matching pokemon to the search query, that way we wouldn't have to perform two loops.

Reply:
Refactored 👍